daemon/daemon.go

```diff
@@ -835,6 +836,13 @@ func (d *Daemon) init() error {
 			return err
 		}
 
+		// Clean all ip-identity entries from BPF map as watcher of key-value
+		// store from which these entries are obtained is started quickly
+		// after boot.
+		if err := remotelxcmap.RemoteLXCMap.DeleteAll(); err != nil {
```

Comment:
I'm not sure this is safe, it means that a restart will temporarily  block all egress traffic. If you really want to do this, then mark the map `NonPersistent` so it will get removed and re-created on opening. It has the benefit that we can change the map format across Cilium versions.

Reply:
Upon startup, maybe we could have the agent set some flag on each value (some mark) that signifies that the entries in the map were there upon agent startup. When the IPIdentityWatcher tries to insert them into the map, it updates the value to not have the mark.  

I can change the IPIdentityWatcher to do an initial `ListPrefix`. Once the initial list from the key-value store is complete, we can then delete all keys that contain this marker by walking over the map again.

Reply:
I've addressed all but this for now.